### PR TITLE
fix: security classification label [WPB-3715]

### PR DESCRIPTION
**Pull request by Garzas**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [ ] contains a reference JIRA issue number like `SQPIT-764`
    - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

- fixed issue that classification label doesn't show when message input field is not open
- refactored security classification banner to have one single composable with view model where it can be reused in OtherUserProfile, OngoingCall and Conversation Screen.
- changed structure of showing banner in OngoingCallScreen by adding spacers to show calling options always in center with or without classification banner

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1119](https://10.10.124.134/job/AR-build-pipeline/1119/) **failed**.

File: app/src/main/kotlin/com/wire/android/ui/calling/ongoing/OngoingCallScreen.kt [hunk 1]
```diff
@@ -310,29 +305,27 @@ private fun OngoingCallTopBar(
...
+    flipCamera: () -> Unit
 ) {
-    Column {
-        val topPadding = if (classificationType != SecurityClassificationType.NONE) {
-            dimensions().spacing8x
-        } else {
-            dimensions().spacing16x
-        }
+    Column(
+        modifier = Modifier.height(96.dp)
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/calling/ongoing/OngoingCallScreen.kt]: suggestion: to make sure it always fits, we should make `SecurityClassificationBanner` also have a strict height, right? And maybe we should extract this value to the `dimensions` to not be hardcoded directly in the code

File: app/src/main/kotlin/com/wire/android/ui/calling/ongoing/OngoingCallScreen.kt [hunk 2]
```diff
@@ -310,29 +305,27 @@ private fun OngoingCallTopBar(
...
+    ) {
+        Spacer(modifier = Modifier.weight(1F))
         Row(
             horizontalArrangement = Arrangement.SpaceEvenly,
             verticalAlignment = Alignment.CenterVertically,
             modifier = Modifier
                 .fillMaxWidth()
-                .padding(top = topPadding)
+                .height(56.dp)
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/calling/ongoing/OngoingCallScreen.kt]: suggestion: same here - maybe take from `dimensions`

File: app/src/main/kotlin/com/wire/android/ui/common/banner/SecurityClassificationArgs.kt [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+import com.wire.kalium.logic.data.id.QualifiedID
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+data class SecurityClassificationArgs(
+    @SerialName("conversation_id") val conversationId: QualifiedID?,
+    @SerialName("user_id") val userId: QualifiedID?
+) : ScopedArgs {
+    override val key = "$ARGS_KEY:${conversationId?.toString().orEmpty()}${userId?.toString().orEmpty()}"
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/common/banner/SecurityClassificationArgs.kt]: suggestion: maybe we should put some spacer in between conversationId and userId, like ":" or "_"

File: app/src/main/kotlin/com/wire/android/ui/common/banner/SecurityClassificationViewModel.kt [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+
+    private var state by mutableStateOf(SecurityClassificationType.NONE)
+
+    override fun state(): SecurityClassificationType = state
+
+    init {
+        when {
+            conversationId != null -> fetchConversationClassificationType(conversationId)
+            userId != null -> fetchUserClassificationType(userId)
+            else -> appLogger.w("Either conversationId or userId should be provided to SecurityClassificationViewModel")
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/common/banner/SecurityClassificationViewModel.kt]: suggestion: maybe we should throw an exception if we don't have both conversationId and userId, because it's an illegal state

File: app/src/main/kotlin/com/wire/android/ui/common/banner/SecurityClassificationArgs.kt [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+import com.wire.android.di.ScopedArgs
+import com.wire.kalium.logic.data.id.QualifiedID
+import kotlinx.serialization.SerialName
+import kotlinx.serialization.Serializable
+
+@Serializable
+data class SecurityClassificationArgs(
+    @SerialName("conversation_id") val conversationId: QualifiedID?,
+    @SerialName("user_id") val userId: QualifiedID?
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/common/banner/SecurityClassificationArgs.kt:28]: suggestion: it's required to have one of them, so maybe we should make an `Either` or disable default constructor and create custom ones to make sure we always create this class with one of them being not null?

File: app/src/main/kotlin/com/wire/android/ui/common/banner/SecurityClassificationViewModel.kt [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+        observeSecurityClassificationLabel.invoke(conversationId)
+            .collect { classificationType ->
+                state = classificationType
+            }
+    }
+
+    private fun fetchUserClassificationType(userId: UserId) = viewModelScope.launch {
+        state = getOtherUserSecurityClassificationLabel(userId)
+    }
+}
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/common/banner/SecurityClassificationViewModel.kt]: question: `getOtherUserSecurityClassificationLabel` just returns a single value and `observeSecurityClassificationLabel` returns a Flow - shouldn't both be unified? Either we get only first value for both cases or we observe for the value changes

File: app/src/main/kotlin/com/wire/android/ui/calling/ongoing/OngoingCallScreen.kt [hunk 7]
```diff
@@ -310,29 +305,27 @@ private fun OngoingCallTopBar(
...
-    Column {
-        val topPadding = if (classificationType != SecurityClassificationType.NONE) {
-            dimensions().spacing8x
-        } else {
-            dimensions().spacing16x
-        }
+    Column(
+        modifier = Modifier.height(96.dp)
+    ) {
+        Spacer(modifier = Modifier.weight(1F))
```
Comment (saleniuk) [inline app/src/main/kotlin/com/wire/android/ui/calling/ongoing/OngoingCallScreen.kt:322]: question: won't this empty space be too big when the security banner is hidden?

File: default.json [hunk 8]
```diff
@@ -14,7 +14,7 @@
             "logging_enabled": true,
             "application_is_private_build": true,
             "development_api_enabled": true,
-            "mls_support_enabled": true,
+            "mls_support_enabled": false,
```
Comment (saleniuk) [inline default.json]: question: is this change intentional? 😅

File: app/src/main/kotlin/com/wire/android/ui/common/banner/SecurityClassificationArgs.kt (hunk 5 above)
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/ui/common/banner/SecurityClassificationArgs.kt:28]: question: also, what will happen if both are not null? 😄

File: default.json (hunk 8 above)
Comment (Garzas) [reply default.json]: No, in federation squad we always forgot to change it back because without it we cannot use app 😅

File: app/src/main/kotlin/com/wire/android/ui/calling/ongoing/OngoingCallScreen.kt (hunk 1 above)
Comment (Garzas) [reply app/src/main/kotlin/com/wire/android/ui/calling/ongoing/OngoingCallScreen.kt]: Thanks! I will use defaultSheetPeekHeight

File: app/src/main/kotlin/com/wire/android/ui/calling/ongoing/OngoingCallScreen.kt (hunk 2 above)
Comment (Garzas) [reply app/src/main/kotlin/com/wire/android/ui/calling/ongoing/OngoingCallScreen.kt]: Ok

File: app/src/main/kotlin/com/wire/android/ui/common/banner/SecurityClassificationViewModel.kt (hunk 4 above)
Comment (Garzas) [reply app/src/main/kotlin/com/wire/android/ui/common/banner/SecurityClassificationViewModel.kt]: I think warning is a better option because we are using it more for federation purposes and we may not catch exception before release 🤔

File: app/src/main/kotlin/com/wire/android/ui/common/banner/SecurityClassificationArgs.kt (hunk 5 above)
Comment (Garzas) [reply app/src/main/kotlin/com/wire/android/ui/common/banner/SecurityClassificationArgs.kt:28]: I think we cannot do that because both conversationId and userId have the same type 🤔 I can try with sealed classes 🤔

File: app/src/main/kotlin/com/wire/android/ui/calling/ongoing/OngoingCallScreen.kt (hunk 7 above)
Comment (Garzas) [reply app/src/main/kotlin/com/wire/android/ui/calling/ongoing/OngoingCallScreen.kt:322]: I will discuss this with Wolfgang

File: app/src/main/kotlin/com/wire/android/ui/common/banner/SecurityClassificationViewModel.kt (hunk 6 above)
Comment (Garzas) [reply app/src/main/kotlin/com/wire/android/ui/common/banner/SecurityClassificationViewModel.kt]: I can return Flow from GetOtherUserSecurityClassificationLabelUseCase, so new kalium PR 😄

File: app/src/main/kotlin/com/wire/android/ui/common/banner/SecurityClassificationArgs.kt (hunk 5 above)
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/ui/common/banner/SecurityClassificationArgs.kt:28]: Ah, okay, then the other option could be to instead do something like:
```
val id: QualifiedID, // non-null
val type: Type.USER // Type.USER or Type.CONVERSATION
```
but it's fine to keep it as it is
😄

File: app/src/main/kotlin/com/wire/android/ui/common/banner/SecurityClassificationViewModel.kt (hunk 4 above)
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/ui/common/banner/SecurityClassificationViewModel.kt]: Ok, I guess it's not a big deal, because if both are null then this banner simply won't be visible, right? It's just that throwing an exception in that case would let us know that someone screwed up in the code faster 😄 
But it was just a suggestion 😉

File: app/src/main/kotlin/com/wire/android/ui/common/banner/SecurityClassificationViewModel.kt (hunk 6 above)
Comment (saleniuk) [reply app/src/main/kotlin/com/wire/android/ui/common/banner/SecurityClassificationViewModel.kt]: I mean, it would be better to have Flow, but if it worked until now for the user with just taking single value, then we can keep it this way for now

**Comment by AndroidBob** [comment]
Build [1158](https://10.10.124.134/job/AR-build-pipeline/1158/) **failed**.

**Comment by AndroidBob** [comment]
Build [1159](https://10.10.124.134/job/AR-build-pipeline/1159/) **failed**.

**Comment by AndroidBob** [comment]
Build [1160](https://10.10.124.134/job/AR-build-pipeline/1160/) **failed**.

**Review by saleniuk — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [1168](https://10.10.124.134/job/AR-build-pipeline/1168/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.4.0(7330851)(7330851)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.4.0(7330851)(7330851)-staging-compat.apk)

**Review by yamilmedina — APPROVED** [review]

**Comment by AndroidBob** [comment]
Build [1171](https://10.10.124.134/job/AR-build-pipeline/1171/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-v4.4.0(7337769)(7337769)-staging-compat.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/staging/compat/com.wire.android-v4.4.0(7337769)(7337769)-staging-compat.apk)
